Remove legacy metrics in MetricRecordingValidatorApiChannel (#6649)
PegaSysEng/artemis: CHANGELOG.md

## Unreleased Changes

### Breaking Changes
- The `validator_beacon_node_published_attestation_total`, `validator_beacon_node_published_aggregate_total`,
  `validator_beacon_node_send_sync_committee_messages_total`, `validator_beacon_node_send_sync_committee_contributions_total`

>   and `validator_beacon_node_published_block_total` metrics have been deprecated in favour of the new `validator_beacon_node_requests_total` metric.
>   The old metrics have been removed. An update to the [Teku Dashboard](https://grafana.com/grafana/dashboards/13457) that uses the new metric is available.

Reviewer: ```suggestion
  and `validator_beacon_node_published_block_total` metrics have been replaced by the new `validator_beacon_node_requests_total` metric. An update to the [Teku Dashboard](https://grafana.com/grafana/dashboards/13457) that uses the new metric is available.
```
